packages/zoe/src/zoeService/installationStorage.js

```diff
@@ -54,80 +65,88 @@ export const makeInstallationStorage = (
    * an instance is started.
    */
 
-  /** @type {InstallBundleID} */
-  const installBundleID = async bundleID => {
-    assert.typeof(bundleID, 'string', `a bundle ID must be provided`);
-    // this waits until someone tells the host application to store the
-    // bundle into the kernel, with controller.validateAndInstallBundle()
-    const bundleCap = await getBundleCapForID(bundleID);
-    // AWAIT
-
-    /** @type {Installation} */
-    // @ts-expect-error cast
-    const installation = makeBundleIDInstallation();
-    installationsBundleCap.init(installation, harden({ bundleCap, bundleID }));
-    return installation;
-  };
-
   /** @type {InstallBundle} */
   const installSourceBundle = async bundle => {
-    assert.typeof(bundle, 'object', 'a bundle must be provided');
+    typeof bundle === 'object' || Fail`a bundle must be provided`;
+    /** @type {Installation} */
+    bundle || Fail`a bundle must be provided`;
     /** @type {Installation} */
     // @ts-expect-error cast
     const installation = makeBundleInstallation(bundle);
-    assert(bundle, 'a bundle must be provided');
     installationsBundle.init(installation, bundle);
     return installation;
   };
 
-  /** @type {InstallBundle} */
-  const installBundle = async allegedBundle => {
-    // Bundle is a very open-ended type and we must decide here
-    // whether to treat it as either a HashBundle or SourceBundle.
-    // So, we cast it down and inspect it.
-    const bundle = /** @type {unknown} */ (harden(allegedBundle));
-    assert.typeof(bundle, 'object', 'a bundle must be provided');
-    assert(bundle !== null, 'a bundle must be provided');
-    const { moduleFormat } = bundle;
-    if (moduleFormat === 'endoZipBase64Sha512') {
-      const { endoZipBase64Sha512 } = bundle;
-      assert.typeof(
-        endoZipBase64Sha512,
-        'string',
-        `bundle endoZipBase64Sha512 must be a string, got ${endoZipBase64Sha512}`,
-      );
-      return installBundleID(`b1-${endoZipBase64Sha512}`);
-    }
-    return installSourceBundle(bundle);
-  };
+  const InstallationStorageI = M.interface('InstallationStorage', {
+    installBundle: M.call(BundleShape).returns(M.promise()),
+    installBundleID: M.call(M.string()).returns(M.promise()),
+    unwrapInstallation: M.callWhen(M.await(InstallationShape)).returns(M.any()),
```

Comment:
If you wish, for the return type, rather than `M.any()` you could mirror the typedef declaration
```js
 * @returns {Promise<{
 *   bundle?: SourceBundle,
 *   bundleCap?: BundleCap,
 *   bundleID?: BundleID,
 *   installation:Installation
 * }>}
```
as
```suggestion
    unwrapInstallation: M.callWhen(M.await(InstallationShape)).returns(M.splitRecord(
      harden({
        installation: InstallationShape,
      }),
      harden({
        bundle: ???,
        bundleCap: ???,
        bundleID: ???
      }),
      harden({}), 
    ),
```
where the `???` are placeholders for whatever these patterns should be. If you do this, you should probably pull it out into a separately named shape.

Reply:
why is the final `harden({})`  included? Is this saying that (contrary to the typescript) these might not be all the fields? Is that different than the default?

Reply:
Opposite. The last argument matches the rest of the record --- the portion not matched by the first two arguments. If the last argument is omitted, it defaults to `M.any()`, which allows the rest of the record to be anything. The `harden({})` says that the rest of the record must be empty, i.e., that there must not be any properties that were not matched by the first two patterns.

Reply:
Looks done. Resolving.